### DISCUSSION TITLE
How to update to Nextcloud 26 now?

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#2208] [Wiki]
If you don't want to wait for AIO getting updated to NC26 in the near future, you can upgrade to Nextcloud 26 already now by following these steps:

1. First, open the aio interface and create a backup and start your containers again after the backup is successful.

2. Afterwards, run via CLI:
```
# Go into the container
sudo docker exec --user www-data -it nextcloud-aio-nextcloud bash
```

3. Now inside the container:

```
# Switch to the beta channel
php occ config:system:set updater.release.channel --value=beta

# Run the update
php updater/updater.phar --no-interaction && php occ app:enable nextcloud-aio --force

# Switch back to the stable channel
php occ config:system:set updater.release.channel --value=stable

# Exit the container
exit
```

4. The above instructions should upgrade you to Nextcloud 26. You might need to follow the steps a second time to reach NC26. If something goes wrong, feel free to restore the backup.

**Comment by monsieurhannes** [comment]
Worked, thank you! 

The only change I noticed is that my background image got reset, the folder color is still the same. Nothing worth thinking about...

**Reply by jplcz** [reply]
Also worked for me. It might be needed to run the updater command at least 2 times in case there are more versions in between (in my case ``v25.0.3-2-gd49ee0d`` to ``25.0.6 RC1``, then ``26.0.1rc1``). You also may want to check configuration warnings reported by admin panel, e.g. running ``php occ db:add-missing-indices`` was required in my case.

**Comment by emelbardis** [comment]
I'm pretty new to AIO. How long do you typically test until releasing with new NC versions?

**Reply by monsieurhannes** [reply]
See the update policy here: https://github.com/nextcloud/all-in-one#update-policy

**Comment by FlattusBlastus** [comment]
I haz a sad. It did it again. (Windows Docker Desktop). Bad things happen after this.
![image](https://user-images.githubusercontent.com/78336592/227775072-73d70fa7-6415-4e57-9592-45e7ca9bccb3.png)

**Reply by szaimen** [reply]
See step 4.

**Reply by FlattusBlastus** [reply]
Restore from my backup... yes

**Comment by Zoey2936** [comment]
Worked for me, but `sysvsem` was missing like mentioned in #1846, added it to `NEXTCLOUD_ADDITIONAL_PHP_EXTENSIONS` and it was fixed. 
But now the updating part in the overview tab, which was hidden before (under nc 25) is shown again, the AIO app is enabled:
![grafik](https://user-images.githubusercontent.com/75573284/228961983-c552640c-7e00-4560-9385-e7e8384ecb22.png)

**Reply by szaimen** [reply]
the updatenotification app will get removed again as soon as the "official" update comes out.

**Reply by Bruceumba** [reply]
Zoey, how exactly did you add sysvsem to the php.ini, if you recall. I can find no reference to it in #1846.

**Reply by Zoey2936** [reply]
> Zoey, how exactly did you add sysvsem to the php.ini, if you recall. I can find no reference to it in #1846.

I used this: https://github.com/nextcloud/all-in-one#how-to-add-php-extensions-permanently-to-the-nextcloud-container

**Reply by szaimen** [reply]
Just wanted to mention that sysvsem will be added by default with the next beta release: https://github.com/nextcloud/all-in-one/pull/2256

**Reply by szaimen** [reply]
This is now released with v4.8.1 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by apoorv569** [comment]
> You can upgrade to Nextcloud 26 now by following these steps:
> 
>     1. First, open the aio interface and create a backup and start your containers again after the backup is successful.
> 
>     2. Afterwards, run via CLI:
> 
> 
> ```
> # Go into the container
> sudo docker exec --user www-data -it nextcloud-aio-nextcloud bash
> ```
> 
>     3. Now inside the container:
> 
> 
> ```
> # Switch to the beta channel
> php occ config:system:set updater.release.channel --value=beta
> 
> # Run the update
> php updater/updater.phar --no-interaction && php occ app:enable nextcloud-aio --force
> 
> # Switch back to the stable channel
> php occ config:system:set updater.release.channel --value=stable
> 
> # Exit the container
> exit
> ```
> 
>     4. The above instructions should upgrade you to Nextcloud 26. You might need to follow the steps a second time to reach NC26. If something goes wrong, feel free to restore the backup.

Is the beta channel safe to use for production server? Or should I stay on whatever the latest AIO version provides?

**Reply by szaimen** [reply]
> Or should I stay on whatever the latest AIO version provides?

This is up to you but the safest way to do things since e.g. in the moment FTS is not ready yet for Nc26.

**Reply by apoorv569** [reply]
> > Or should I stay on whatever the latest AIO version provides?
> 
> This is up to you but the safest way to do things since e.g. in the moment FTS is not ready yet for Nc26.

I see. I think I will wait for the stable channel to receive the latest update.

**Reply by SuperDarius-git** [reply]
Just before I update to 26, will this work on latest-ARM64? Should I also use the Beta channel? Just want to make sure...

**Comment by Croydon** [comment]
I really appreciate all your work on AIO. It is really useful!

That said, my biggest frustration point with AIO in general is its in-transparency: Why is AIO doing things the way it is? And how are you guys deciding things for further changes?

This is both a problem here on GitHub where changes are oftentimes not explained and no context is given or how the decisions were made, and also a problem in AIO itself. 

In the recent AIO we are suddenly getting this notice about Nextcloud 26 with a link here. Here we are getting a list of instructions how to update to NC26, but also no further explanations whatsoever.

This raises many questions, including:
  1. Do I have to do this manually sooner or later, or will it happen with the regular update process of AIO automatically at some point?
  2. Would this be considered an early adoption or late adoption?
  3. Is this safe to do?
  4. Is it recommended?
  5. Why this notice now, when the previous NC upgrade via AIO did not have such a notice?


Explaining things, features, changes more would really help consumers and boost confidence in AIO.

**Reply by szaimen** [reply]
I think you are overreacting here. We simply wanted to make things more verbose by documenting a way to upgrade already now.

I adjusted the sentence at the top to make things a bit clearer:

> If you don't want to wait for AIO getting updated to NC26 in the near future, you can upgrade to Nextcloud 26 already now by following these steps:

**Reply by fatth0r** [reply]
@Croydon, I think the underlying problem is you don't understand what AIO is to begin with.  Whereas Nextcloud is a piece of software, Nextcloud AIO is a turn-key solution.  AIO is an integration of the Nextcloud software suite (of course), plus Nextcloud Talk, Collabora, a database, a backup tool, ClamAV, plus tons of other tools and features.

Why do AIO releases lag slightly behind Nextcloud releases?  That's because config management, software integration, and quality testing are complex tasks that take time for largely volunteer teams to complete.  If you're unhappy with the development practices of this free tool, maybe you'd like to 1) volunteer your time to the project, 2) financially support the project, or 3) fork the project and develop a variant that better meets your needs.

**Reply by Croydon** [reply]
I just tried to provide feedback on the user experience AIO has, which in my humble opinion can be easily improved by explaining features and decisions better. I felt like this for a longer time, and just decided to use this recent update notice as an example to try to make my point clear.

Such small sentences like @szaimen added above are sometimes already enough to fix the situation and improve the UX dramatically. Consumers should never have to guess what an application does or expects from them.

I understand what AIO is, thank you. I also understand that open source work is hard, there is never enough time to address everything and oftentimes people are not getting paid at all for their work. All I tried is providing useful feedback.

**Reply by szaimen** [reply]
I'll gladly improve the documentation or details if you provide specific requests. Just guessing what might not be good enough documented is unfortunately not really possible.

**Reply by szaimen** [reply]
Even more welcome would be PRs that help to improve the details :)

**Comment by pktiuk** [comment]
I have a question related to this kind of updates.  
Is there any way of checking which of the installed apps are compatible with the latest version of Nextcloud?

**Reply by LeLuc4** [reply]
Maybe this is not the place to ask this. But you can see it directly in the store. E.g. https://apps.nextcloud.com/apps/drawio has 26 support, but https://apps.nextcloud.com/apps/rainloop has not.

**Reply by pktiuk** [reply]
I was looking for a bit more automated way, because checking every of my 30 apps manually is a bit annoying.   
(but I guess there is no other way :/ )
